microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/SubsystemDefinition.java

```diff
@@ -79,4 +114,39 @@ public void registerAdditionalRuntimePackages(final ManagementResourceRegistrati
             resourceRegistration.registerAdditionalRuntimePackages(RuntimePackageDependency.required(m));
         }
     }
+
+    private static final class DefaultTracerWriteAttributeHandler extends ReloadRequiredWriteAttributeHandler {
+
+        private DefaultTracerWriteAttributeHandler() {
+            super(DEFAULT_TRACER);
+        }
+
+        @Override
+        protected void recordCapabilitiesAndRequirements(OperationContext context, AttributeDefinition attributeDefinition,
+                ModelNode newValue, ModelNode oldValue) {
+            super.recordCapabilitiesAndRequirements(context, attributeDefinition, newValue, oldValue);
+            if (oldValue.isDefined()) {
+                context.deregisterCapability(DEFAULT_TRACER_CAPABILITY.getDynamicName(oldValue.asString()));
+            }
+            if (newValue.isDefined()) {
+                context.registerCapability(RuntimeCapability.Builder.of(DEFAULT_TRACER_CAPABILITY.getDynamicName(newValue.asString()), false).build());
+            }
+        }
+    }
+
+    private static final class SubsystemRemoveHandler extends ReloadRequiredRemoveStepHandler {
+
+        private SubsystemRemoveHandler() {
+            super();
+        }
+
+        @Override
+        protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+            super.recordCapabilitiesAndRequirements(context, operation, resource);
```

Comment:
I don't think you want to call the super implementation, as you are handling capability registration conditionally.